Code/PgSQL/rdkit/adapter.cpp

```diff
@@ -594,7 +595,13 @@ extern "C" int MolSubstruct(CROMol i, CROMol a, bool useChirality) {
     params.useEnhancedStereo = getDoEnhancedStereoSSS();
   }
   params.maxMatches = 1;
-  auto matchVect = RDKit::SubstructMatch(*im, *am, params);
+
+  if (useMatchers) {
+    GenericGroups::setGenericQueriesFromProperties(*am);
```

Comment:
Have you timed this running on a querying against a large database?
I'm somewhat concerned that calling this on the query molecule for every molecule in the database will make things slow.
Maybe that's not actually the case though?

Reply:
We have been using that on production and didn't notice much slowdown. But that was one of my worries and why I didn't make it the default.
Note that are not querying the full database anymore as we are slicing it by "user" so it may not be noticeable on just millions of structures?

Reply:
But you are right maybe we should do that once only.

Reply:
Ok the last commit should fix that.

Reply:
> Ok the last commit should fix that.

This isn't something I'm certain about, but I don't think that last change would have any impact on how often that function is called. I think in order to "fix" that (assuming it's even a problem, which it sounds like it may not be) there would need to be either a function at the SQL level that one calls in advance or a property added to the molecule tracking whether or not it's been transformed (and we'd then need to check that everything actually gets preserved when the molecule is serialized... atom and mol properties are not always preserved).

If you haven't observed any problems on million+ structure queries, I think we don't need to worry about it for now. Can always come back to it later if need be.

If you agree with this, it would be nice if you could revert your last commit... the code was easier to follow beorehand.

Reply:
Reverted.

I'll try to benchmark it and see if it is worth adding a new SQL function for converting the query before hand.

Reply:
The reason I didn't do it initially was because I couldn't get it serialized and get this parameter. But I only tried from the Python side when sending the query as a binary.

Reply:
> The reason I didn't do it initially was because I couldn't get it serialized and get this parameter. But I only tried from the Python side when sending the query as a binary.

Yeah, the properties are, for space and efficiency reasons, not pickled by default. You can change this though:
```
In [7]: m = Chem.MolFromSmarts('CO* |$;;ARY_p$|')

In [8]: Chem.SetGenericQueriesFromProperties(m)

In [9]: m.GetAtomWithIdx(2).GetPropsAsDict()
Out[9]: {'_QueryAtomGenericLabel': 'ARY'}

In [10]: m2 = Chem.Mol(m.ToBinary())

In [11]: m2.GetAtomWithIdx(2).GetPropsAsDict()
Out[11]: {}

In [12]: me = Chem.Mol(m.ToBinary(Chem.PropertyPickleOptions.AllProps))

In [13]: me.GetAtomWithIdx(2).GetPropsAsDict()
Out[13]: {'_QueryAtomGenericLabel': 'ARY'}
```

Reply:
So if we can do that, does it make sense to still do it in the cartridge?

Reply:
I think there's value in having the cartridge be self-contained